cvmfs/network/download.cc

```diff
@@ -1667,6 +1770,13 @@ DownloadManager::~DownloadManager() {
     health_check_.Reset();
   }
 
+  int32_t old_count =
+                    atomic_xadd32(&DownloadManager::counter_use_data_tube_, -1);
+  if (old_count == 1) {
+    delete data_tube_empty_elements_;
```

Comment:
This breaks the test suite because the above count hits zero and this object is deleted. But then a subsequent test is spawned and it reads freed memory. So either no delete or you need to re-allocate on demand.